MiniApp/Classes/API/ListingAPI.swift

```diff
@@ -1,21 +1,47 @@
 internal class ListingApi {
     let environment: Environment
+    var path: String {
+        return "/oneapp/ios/\(environment.appVersion)/miniapps"
+    }
 
     init(environment: Environment) {
         self.environment = environment
     }
 
     func createURLRequest() -> URLRequest? {
+
         guard let url = getListingURL() else {
             return nil
         }
         return URLRequest.createURLRequest(url: url, environment: environment)
     }
 
+    func createURLRequest(for miniAppID: String) -> URLRequest? {
+
+        guard let url = getListingURL(for: miniAppID) else {
+            return nil
+        }
+        return URLRequest.createURLRequest(url: url, environment: environment)
+    }
+
     private func getListingURL() -> URL? {
         guard let baseURL = environment.baseUrl else {
             return nil
         }
-        return baseURL.appendingPathComponent("/oneapp/ios/\(environment.appVersion)/miniapps")
+        return baseURL.appendingPathComponent(path)
+    }
+
+    private func getListingURL(for miniAppId: String) -> URL? {
+        guard let baseURL = environment.baseUrl else {
+            return nil
+        }
```

Comment:
Maybe we can move this into some function to avoid duplicates?

Reply:
yes, got it